src/iris/sender/coordinator.py

```diff
@@ -0,0 +1,106 @@
+import logging
+from iris import db, metrics
+from gevent import sleep
+from itertools import cycle
+
+logger = logging.getLogger(__name__)
+
+UPDATE_FREQUENCY = 3
+SENDER_ALIVE_TIMEOUT = 10
+
+GET_MASTER_QUERY = '''SELECT `sender_address` FROM `sender_master_election` WHERE `anchor` = 1'''
+
+GET_SLAVES_QUERY = '''
+  SELECT `sender_address`
+  FROM `sender_instances`
+  WHERE `last_seen` > NOW() - INTERVAL CAST(:timeout AS UNSIGNED) SECOND
+  AND `sender_address` NOT IN (%s)''' % GET_MASTER_QUERY
+
+UPDATE_MASTER_QUERY = '''
+  INSERT IGNORE INTO `sender_master_election` (`anchor`, `sender_address`, `last_seen_active`) VALUES (
+    1, :me, NOW()
+  ) ON DUPLICATE KEY UPDATE
+    `sender_address` = if(`last_seen_active` < NOW() - INTERVAL CAST(:timeout AS UNSIGNED) SECOND, VALUES(`sender_address`), `sender_address`),
+    `last_seen_active` = if(`sender_address` = VALUES(`sender_address`), VALUES(`last_seen_active`), `last_seen_active`)
+'''
+
+UPDATE_INSTANCES_QUERY = '''
+  INSERT INTO `sender_instances` (`sender_address`, `last_seen`) VALUES(:me, now())
+  ON DUPLICATE KEY UPDATE `last_seen` = NOW()'''
+
+
+class Coordinator(object):
+    def __init__(self, hostname, port):
+        self.me = '%s:%s' % (hostname, port)
+        self.is_master = None
+        self.master = None
+
+    def am_i_master(self):
+        return self.is_master
+
+    def get_current_master(self):
+        session = db.Session()
+        master = session.execute(GET_MASTER_QUERY).scalar()
```

Comment:
If we want to keep `self.master`, then I suggest we remove `self.is_master`, so it's one less variable that we need to keep it up to date. We can replace all `self.is_master` with `self.me == self.master`.